src/model/calculations.ts

```diff
@@ -8,19 +8,25 @@ function convertRatioToDecimalNumber(ratio: {
   return quotient;
 }
 
-// Reference: https://docs.google.com/document/d/19iVTJfJxIMr_gQHzMgSAHR6PhCBFEHg0TahjYc8uOac/edit#heading=h.fiexe0ua8jk7
-export class OutstandingStockSharesCalculator {
-  private value_: Big = Big("0");
-  private issuanceAmounts_: Map<string, string> = new Map();
-  private reissuedSecurityIds_: Set<string> = new Set();
+interface Transaction {
+  object_type: string;
+  security_id: string;
+  quantity?: string;
+  quantity_converted?: string;
+}
+
+abstract class OutstandingEquityCalculatorBase {
+  protected value_: Big = Big("0");
+  protected issuanceAmounts_: Map<string, string> = new Map();
+  protected pendingSecurityIds_: Set<string> = new Set();
```

Comment:
**Non-blocking:** Let's think about a better name for `pendingSecurityIds_` as we proceed. It's not really the right name.